### ISSUE TITLE
Path error

### ISSUE THREAD

**Issue by konstantin-921** [corazawaf/coraza-caddy#5]
First of all I want to say thank you for the alternative to products like ModSecurity. This is very important for those who use open source. Before that, I had hardly ever met Go. That's why even simple errors can be difficult for me.

Version Go: 1.17
OS: Ubuntu 18.04

```caddy build-info``` output:
```
go_version: go1.17.6
go_os:      linux
go_arch:    amd64
path:       caddy
main:       caddy (devel) 
dependencies:
github.com/AndreasBriese/bbloom v0.0.0-20190825152654-46b345b51c96 h1:cTp8I5+VIoKjsnZuH8vjyaysT/ses3EvZeaV/1UkF2M=
github.com/Masterminds/goutils v1.1.1 h1:5nUrii3FMTL5diU80unEVvNevw1nH4+ZV4DSLVJLSYI=
github.com/Masterminds/semver/v3 v3.1.1 h1:hLg3sBzpNErnxhQtUy/mmLR2I9foDujNK030IGemrRc=
github.com/Masterminds/sprig/v3 v3.2.2 h1:17jRggJu518dr3QaafizSXOjKYp94wKfABxUmyxvxX8=
github.com/alecthomas/chroma v0.9.2 h1:yU1sE2+TZbLIQPMk30SolL2Hn53SR/Pv750f7qZ/XMs=
github.com/antchfx/jsonquery v1.1.4 h1:+OlFO3QS9wjU0MKx9MgHm5f6o6hdd4e9mUTp0wTjxlM=
github.com/antchfx/xmlquery v1.3.6 h1:kaEVzH1mNo/2AJZrhZjAaAUTy2Nn2zxGfYYU8jWfXOo=
github.com/antchfx/xpath v1.2.0 h1:mbwv7co+x0RwgeGAOHdrKy89GvHaGvxxBtPK0uF9Zr8=
github.com/antlr/antlr4 v0.0.0-20200503195918-621b933c7a7f h1:0cEys61Sr2hUBEXfNV8eyQP01oZuBgoMeHunebPirK8=
github.com/beorn7/perks v1.0.1 h1:VlbKKnNfV8bJzeqoa4cOKqO6bYr3WgKZxO8Z16+hsOM=
github.com/caddyserver/caddy/v2 v2.4.6 h1:HGkGICFGvyrodcqOOclHKfvJC0qTU7vny/7FhYp9hNw=
github.com/caddyserver/certmagic v0.15.2 h1:OMTakTsLM1ZfzMDjwvYprfUgFzpVPh3u87oxMPwmeBc=
github.com/cespare/xxhash v1.1.0 h1:a6HrQnmkObjyL+Gs60czilIUGqrzKutQD6XZog3p+ko=
github.com/cespare/xxhash/v2 v2.1.1 h1:6MnRN8NT7+YBpUIWxHtefFZOKTAPgGjpQSxqLNn0+qY=
github.com/cheekybits/genny v1.0.0 h1:uGGa4nei+j20rOSeDeP5Of12XVm7TGUd4dJA9RDitfE=
github.com/chzyer/readline v0.0.0-20180603132655-2972be24d48e h1:fY5BOSpyZCqRo5OhCuC+XN+r/bBCmeuuJtjz+bCNIf8=
github.com/cpuguy83/go-md2man/v2 v2.0.0 h1:EoUDS0afbrsXAZ9YQ9jdu/mZ2sXgT1/2yyNng4PGlyM=
github.com/danwakefield/fnmatch v0.0.0-20160403171240-cbb64ac3d964 h1:y5HC9v93H5EPKqaS1UYVg1uYah5Xf51mBfIoWehClUQ=
github.com/dgraph-io/badger v1.6.2 h1:mNw0qs90GVgGGWylh0umH5iag1j6n/PeJtNvL6KY/x8=
github.com/dgraph-io/badger/v2 v2.2007.4 h1:TRWBQg8UrlUhaFdco01nO2uXwzKS7zd+HVdwV/GHc4o=
github.com/dgraph-io/ristretto v0.0.4-0.20200906165740-41ebdbffecfd h1:KoJOtZf+6wpQaDTuOWGuo61GxcPBIfhwRxRTaTWGCTc=
github.com/dgryski/go-farm v0.0.0-20200201041132-a6ae2369ad13 h1:fAjc9m62+UWV/WAFKLNi6ZS0675eEUC9y3AlwSbQu1Y=
github.com/dlclark/regexp2 v1.4.0 h1:F1rxgk7p4uKjwIQxBs9oAXe5CqrXlCduYEJvrF4u93E=
github.com/dustin/go-humanize v1.0.1-0.20200219035652-afde56e7acac h1:opbrjaN/L8gg6Xh5D04Tem+8xVcz6ajZlGCs49mQgyg=
github.com/go-chi/chi v4.1.2+incompatible h1:fGFk2Gmi/YKXk0OmGfBh0WgmN3XB8lVnEyNz34tQRec=
github.com/go-kit/kit v0.10.0 h1:dXFJfIHVvUcpSgDOV+Ne6t7jXri8Tfv2uOLHUZ2XNuo=
github.com/go-logfmt/logfmt v0.5.0 h1:TrB8swr/68K7m9CcGut2g3UOihhbcbiMAYiuTXdEih4=
github.com/go-sql-driver/mysql v1.6.0 h1:BCTh4TKNUYmOmMUcQ3IipzF5prigylS7XXjEkfCHuOE=
github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da h1:oI5xCqsCo564l8iNU+DwB5epxmsaqB+rhGL0m5jtYqE=
github.com/golang/protobuf v1.5.2 h1:ROPKBNFfQgOUMifHyP+KYbvpjbdoFNs+aK7DXlji0Tw=
github.com/golang/snappy v0.0.3 h1:fHPg5GQYlCeLIPB9BZqMVR5nR9A+IM5zcgeTdjMYmLA=
github.com/google/cel-go v0.7.3 h1:8v9BSN0avuGwrHFKNCjfiQ/CE6+D6sW+BDyOVoEeP6o=
github.com/google/uuid v1.3.0 h1:t6JiXgmwXMjEs8VusXIJk2BXHsn+wx8BZdTaoZ5fu7I=
github.com/huandu/xstrings v1.3.2 h1:L18LIDzqlW6xN2rEkpdV8+oL/IXWJ1APd+vsdYy4Wdw=
github.com/imdario/mergo v0.3.12 h1:b6R2BslTbIEToALKP7LxUvijTsNI9TAe80pLWN2g/HU=
github.com/ip2location/ip2location-go/v9 v9.1.0 h1:DVlKxIcjA7CsA0cgzbidwe6eKzpbkS313LsH9ceutxI=
github.com/jptosso/aho-corasick v1.0.4 h1:4zLohlv0YrkXNLhNdRA0eTanzd1z8YFnDfZYg0fTm9g=
github.com/jptosso/coraza-caddy v1.1.0 h1:VR5P1VEiF3K5fvAjnZIpLpniW7eVmbAislI3G5ZfWNs=
github.com/jptosso/coraza-waf v1.1.0 h1:tItgbuhNXdigvwpRHzQkudaVrG9G9QNNFArajz5yFbQ=
github.com/juju/ansiterm v0.0.0-20180109212912-720a0952cc2a h1:FaWFmfWdAUKbSCtOU2QjDaorUexogfaMgbipgYATUMU=
github.com/klauspost/compress v1.13.6 h1:P76CopJELS0TiO2mebmnzgWaajssP/EszplttgQxcgc=
github.com/klauspost/cpuid/v2 v2.0.9 h1:lgaqFMSdTdQYdZ04uHyN2d/eKdOMyi2YLSvlQIBFYa4=
github.com/libdns/libdns v0.2.1 h1:Wu59T7wSHRgtA0cfxC+n1c/e+O3upJGWytknkmFEDis=
github.com/lucas-clemente/quic-go v0.23.0 h1:5vFnKtZ6nHDFsc/F3uuiF4T3y/AXaQdxjUqiVw26GZE=
github.com/lunixbochs/vtclean v1.0.0 h1:xu2sLAri4lGiovBDQKxl5mrXyESr3gUr5m5SM5+LVb8=
github.com/manifoldco/promptui v0.8.0 h1:R95mMF+McvXZQ7j1g8ucVZE1gLP3Sv6j9vlF9kyRqQo=
github.com/marten-seemann/qpack v0.2.1 h1:jvTsT/HpCn2UZJdP+UUB53FfUUgeOyG5K1ns0OJOGVs=
github.com/marten-seemann/qtls-go1-17 v0.1.0 h1:P9ggrs5xtwiqXv/FHNwntmuLMNq3KaSIG93AtAZ48xk=
github.com/mattn/go-colorable v0.1.8 h1:c1ghPdyEDarC70ftn0y+A/Ee++9zz8ljHG1b13eJ0s8=
github.com/mattn/go-isatty v0.0.13 h1:qdl+GuBjcsKKDco5BsxPJlId98mSWNKqYA+Co0SC1yA=
github.com/matttproud/golang_protobuf_extensions v1.0.1 h1:4hp9jkHxhMHkqkrB3Ix0jegS5sx/RkqARlsWZ6pIwiU=
github.com/mgutz/ansi v0.0.0-20200706080929-d51e80ef957d h1:5PJl274Y63IEHC+7izoQE9x6ikvDFZS2mDVS3drnohI=
github.com/mholt/acmez v1.0.1 h1:J7uquHOKEmo71UDnVApy1sSLA0oF/r+NtVrNzMKKA9I=
github.com/micromdm/scep/v2 v2.1.0 h1:2fS9Rla7qRR266hvUoEauBJ7J6FhgssEiq2OkSKXmaU=
github.com/miekg/dns v1.1.43 h1:JKfpVSCB84vrAmHzyrsxB5NAr5kLoMXZArPSw7Qlgyg=
github.com/mitchellh/copystructure v1.2.0 h1:vpKXTN4ewci03Vljg/q9QvCGUDttBOGBIa15WveJJGw=
github.com/mitchellh/reflectwalk v1.0.2 h1:G2LzWKi524PWgd3mLHV8Y5k7s6XUvT0Gef6zxSIeXaQ=
github.com/naoina/go-stringutil v0.1.0 h1:rCUeRUHjBjGTSHl0VC00jUPLz8/F9dDzYI70Hzifhks=
github.com/naoina/toml v0.1.1 h1:PT/lllxVVN0gzzSqSlHEmP8MJB4MY2U7STGxiouV4X8=
github.com/oschwald/geoip2-golang v1.5.0 h1:igg2yQIrrcRccB1ytFXqBfOHCjXWIoMv85lVJ1ONZzw=
github.com/oschwald/maxminddb-golang v1.8.0 h1:Uh/DSnGoxsyp/KYbY1AuP0tYEwfs0sCph9p/UMXK/Hk=
github.com/pcktdmp/cef v0.2.0 h1:MgFUXCRqaXsITq83bN0+mmXVh4CGFgOeuyxNaT86Kj0=
github.com/pkg/errors v0.9.1 h1:FEBLx1zS214owpjy7qsBeixbURkuhQAwrK5UwLGTwt4=
github.com/prometheus/client_golang v1.11.0 h1:HNkLOAEQMIDv/K+04rukrLx6ch7msSRwf3/SASFAGtQ=
github.com/prometheus/client_model v0.2.0 h1:uq5h0d+GuxiXLJLNABMgp2qUWDPiLvgCzz2dUR+/W/M=
github.com/prometheus/common v0.26.0 h1:iMAkS2TDoNWnKM+Kopnx/8tnEStIfpYA0ur0xQzzhMQ=
github.com/prometheus/procfs v0.6.0 h1:mxy4L2jP6qMonqmq+aTtOx1ifVWUgG/TAmntgbh3xv4=
github.com/rs/xid v1.2.1 h1:mhH9Nq+C1fY2l1XIpgxIiUOfNpRBYH1kKcr+qfKgjRc=
github.com/russross/blackfriday/v2 v2.0.1 h1:lPqVAte+HuHNfhJ/0LC98ESWRz8afy9tM/0RK8m9o+Q=
github.com/samfoo/ansi v0.0.0-20160124022901-b6bd2ded7189 h1:CmSpbxmewNQbzqztaY0bke1qzHhyNyC29wYgh17Gxfo=
github.com/shopspring/decimal v1.2.0 h1:abSATXmQEYyShuxI4/vyW3tV1MrKAJzCZ/0zLUXYbsQ=
github.com/shurcooL/sanitized_anchor_name v1.0.0 h1:PdmoCO6wvbs+7yrJyMORt4/BmY5IYyJwS/kOiWx8mHo=
github.com/sirupsen/logrus v1.8.1 h1:dJKuHgqk1NNQlqoA6BTlM1Wf9DOH3NBjQyu0h9+AZZE=
github.com/smallstep/certificates v0.17.5-0.20211008195551-04fe3126bebf h1:T27FAcJuIadMwnt5uaYCxeNQceUoCkLHqWc0Y+PneHE=
github.com/smallstep/cli v0.17.6 h1:0npb8eQGDgEBPziYXb9tW4f4S5jt8aRRmEJvs97oh+Y=
github.com/smallstep/nosql v0.3.8 h1:1/EWUbbEdz9ai0g9Fd09VekVjtxp+5+gIHpV2PdwW3o=
github.com/smallstep/truststore v0.9.6 h1:vNzEJmaJL0XOZD8uouXLmYu4/aP1UQ/wHUopH3qKeYA=
github.com/spf13/cast v1.4.1 h1:s0hze+J0196ZfEMTs80N7UlFt0BDuQ7Q+JDnHiMWKdA=
github.com/stoewer/go-strcase v1.2.0 h1:Z2iHWqGXH00XYgqDmNgQbIBxf3wrNq0F3feEy0ainaU=
github.com/urfave/cli v1.22.5 h1:lNq9sAHXK2qfdI8W+GRItjCEkI+2oR4d+MEHy1CKXoU=
github.com/yuin/goldmark v1.4.1 h1:/vn0k+RBvwlxEmP5E7SZMqNxPhfMVFEJiykr15/0XKM=
github.com/yuin/goldmark-highlighting v0.0.0-20210516132338-9216f9c5aa01 h1:0SJnXjE4jDClMW6grE0xpNhwpqbPwkBTn8zpVw5C0SI=
go.etcd.io/bbolt v1.3.6 h1:/ecaJf0sk1l4l6V4awd65v2C3ILy7MSj+s/x1ADCIMU=
go.mozilla.org/pkcs7 v0.0.0-20210826202110-33d05740a352 h1:CCriYyAfq1Br1aIYettdHZTy8mBTIPo7We18TuO/bak=
go.step.sm/cli-utils v0.6.0 h1:sH4FxBcjmbxyilKXheSyJuKF/QjpojpiW90ERwUWOgQ=
go.step.sm/crypto v0.11.0 h1:VDpeVgEmqme/FK2w5QINxkOQ1FWOm/Wi2TwQXiacKr8=
go.step.sm/linkedca v0.5.0 h1:oZVRSpElM7lAL1XN2YkjdHwI/oIZ+1ULOnuqYPM6xjY=
go.uber.org/atomic v1.9.0 h1:ECmE8Bn/WFTYwEW/bpKD3M8VtR/zQVbavAoalC1PYyE=
go.uber.org/multierr v1.7.0 h1:zaiO/rmgFjbmCXdSYJWQcdvOCsthmdaHfr3Gm2Kx4Ec=
go.uber.org/zap v1.19.0 h1:mZQZefskPPCMIBCSEH0v2/iUqqLrYtaeqwD6FUGUnFE=
golang.org/x/crypto v0.0.0-20210915214749-c084706c2272 h1:3erb+vDS8lU1sxfDHF4/hhWyaXnhIaO+7RgL4fDZORA=
golang.org/x/net v0.0.0-20210913180222-943fd674d43e h1:+b/22bPvDYt4NPDcy4xAGCmON713ONAWFeY3Z7I3tR8=
golang.org/x/sys v0.0.0-20210915083310-ed5796bab164 h1:7ZDGnxgHAMw7thfC5bEos0RDAccZKxioiWBhfIe+tvw=
golang.org/x/term v0.0.0-20210503060354-a79de5458b56 h1:b8jxX3zqjpqb2LklXPzKSGJhzyxCOZSz8ncv8Nv+y7w=
golang.org/x/text v0.3.7 h1:olpwvP2KacW1ZWvsR7uQhoyTYvKAupfQrRGBFM352Gk=
google.golang.org/genproto v0.0.0-20210719143636-1d5a45f8e492 h1:7yQQsvnwjfEahbNNEKcBHv3mR+HnB1ctGY/z1JXzx8M=
google.golang.org/grpc v1.39.0 h1:Klz8I9kdtkIN6EpHHUOMLCYhTn/2WAe5a0s1hcBkdTI=
google.golang.org/protobuf v1.27.1 h1:SnqbnDw1V7RiZcXPx5MEeqPv2s79L9i7BJUlG/+RurQ=
gopkg.in/natefinch/lumberjack.v2 v2.0.0 h1:1Lc07Kr7qY4U2YPouBjpCLxpiyxIVoxqXgkXLknAOE8=
gopkg.in/square/go-jose.v2 v2.6.0 h1:NGk74WTnPKBNUhNzQX7PYcTLUjoq7mzKk2OKbvwk2iI=
gopkg.in/yaml.v2 v2.4.0 h1:D8xgwECY7CYvx+Y2n4sBz93Jn9JRvxdiyyo8CTfuKaY=
```

### How I install Caddy + Coraza WAF

1. I added systemd service and built Caddy using this command:
```
xcaddy build --with github.com/jptosso/coraza-caddy@latest=github.com/konstantin-921/coraza-caddy@latest
```
github.com/konstantin-921/coraza-caddy@latest - it is my fork with uncommented plugins ```github.com/jptosso/coraza-libinjection``` and ```github.com/coraza-pcre``` from caddy/main.go

2. Downloaded CSR and put it in caddy dir (you can find paths in include section)

3. Added following settings to Caddyfile:
```
{
	debug
	auto_https off
	log {
		output file /var/log/caddy-access.log
	}
	order coraza_waf first
}

example.domain.com {
	coraza_waf {
		directives `
		SecAction "id:1,pass,log"
		`
		include /etc/caddy/coraza.conf-recommended
		include /etc/caddy/coreruleset/crs-setup.conf.example
		include /etc/caddy/coreruleset/rules/*.conf
	}
	tls /etc/caddy/domain.crt /etc/caddy/domain.key
	log {
		output file /var/log/caddy-access.log
	}
	respond "Hello HTTPS from example.domain.com!"
}
```

4. Enabled ```SecRuleEngine On```

5. Started Caddy like service ```systemctl start caddy```

### Problem

My problem is that I get this error when I add includes to Caddyfile:
```
caddy[12696]: {"level":"error","ts":1642752737.1271152,"msg":"cannot read configurations file","path":"/etc/caddy/coreruleset/rules/*.conf"}
caddy[13844]: run: loading initial config: loading new config: loading http app module: provision http: server srv0: setting up route handlers: route 0: loading handler modules: position 0: loading module 'subroute': provision http.handlers.subroute: setting up subroutes: route 0: loading handler modules: position 0: loading module 'waf': provision http.handlers.waf: cannot load waf directives invalid profile path
```

In rules folder I have all files from this folder - https://github.com/coreruleset/coreruleset/tree/v3.4/dev/rules. All of the paths look right. I don't understand what's wrong.

Caddy can be started if I specify a specific file. For example, like this:
```
include /etc/caddy/crs/rules/REQUEST-930-APPLICATION-ATTACK-LFI.conf
```
But the Coraza itself apparently does not work because my requests with LFI vulnerabilities are freely handled by Caddy although they should be blocked by Coraza.

**Comment by konstantin-921** [comment]
It seems to be CRS, because the following configuration works and prevents URI /test5, /test6
```
{
    debug
    auto_https off
    log {
        output file /var/log/caddy-access.log
    }
    order coraza_waf first
}

example.domain.com {
    coraza_waf {
        directives `
            SecAction "id:1,pass,log"
            SecRule REQUEST_URI "/test5" "id:2, deny, log, phase:1"
            SecRule REQUEST_URI "/test6" "id:4, deny, log, phase:3"
        `
        #include /etc/caddy/coraza.conf-recommended
        #include /etc/caddy/coreruleset/crs-setup.conf.example
        #include /etc/caddy/coreruleset/rules/*.conf
    }
    tls /etc/caddy/surelc.crt /etc/caddy/surelc.key
    log {
        output file /var/log/caddy-access.log
    }
    respond "Hello HTTPS from example.domain.com!"
}

```

**Comment by jptosso** [comment]
He @konstantin-921 I think you can use the xcaddy compiler to include coraza plugins, you could try -with github.com/jptosso/coraza-pcre

About the issue, it sounds like a bug, actually www.tosso.io is running the latest version of coraza-caddy but when you trigger an error all pages will return a 403 error for a few minutes.

Thank you for the information, once I will keep you posted once I find the cause and fix.

**Comment by konstantin-921** [comment]
@jptosso Thanks for feedback. I'll follow your advice about using xcaddy. But as I understand this is unlikely to solve my main problem.

**Comment by konstantin-921** [comment]
I tried to build Caddy using this command:
```
xcaddy build --with github.com/jptosso/coraza-caddy --with github.com/jptosso/coraza-pcre --with github.com/jptosso/coraza-libinjection
```

Got an error:

```
root@nginx-test3-fl:~# xcaddy build --with github.com/jptosso/coraza-caddy github.com/jptosso/coraza-pcre github.com/jptosso/coraza-libinjection
2022/01/22 04:09:52 [ERROR] missing flag; caddy version already set at github.com/jptosso/coraza-pcre
root@nginx-test3-fl:~# xcaddy build --with github.com/jptosso/coraza-caddy --with github.com/jptosso/coraza-pcre --with github.com/jptosso/coraza-libinjection
2022/01/22 04:10:13 [INFO] Temporary folder: /tmp/buildenv_2022-01-22-0410.3712383977
2022/01/22 04:10:13 [INFO] Writing main module: /tmp/buildenv_2022-01-22-0410.3712383977/main.go
2022/01/22 04:10:13 [INFO] Initializing Go module
2022/01/22 04:10:13 [INFO] exec (timeout=10s): /usr/local/go/bin/go mod init caddy 
go: creating new go.mod: module caddy
go: to add module requirements and sums:
	go mod tidy
2022/01/22 04:10:13 [INFO] Pinning versions
2022/01/22 04:10:13 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddyserver/caddy/v2 
go get: added github.com/beorn7/perks v1.0.1
go get: added github.com/caddyserver/caddy/v2 v2.4.6
go get: added github.com/caddyserver/certmagic v0.15.2
go get: added github.com/cespare/xxhash/v2 v2.1.1
go get: added github.com/golang/protobuf v1.5.2
go get: added github.com/google/uuid v1.3.0
go get: added github.com/klauspost/cpuid/v2 v2.0.9
go get: added github.com/libdns/libdns v0.2.1
go get: added github.com/matttproud/golang_protobuf_extensions v1.0.1
go get: added github.com/mholt/acmez v1.0.1
go get: added github.com/miekg/dns v1.1.43
go get: added github.com/prometheus/client_golang v1.11.0
go get: added github.com/prometheus/client_model v0.2.0
go get: added github.com/prometheus/common v0.26.0
go get: added github.com/prometheus/procfs v0.6.0
go get: added go.uber.org/atomic v1.7.0
go get: added go.uber.org/multierr v1.6.0
go get: added go.uber.org/zap v1.19.0
go get: added golang.org/x/crypto v0.0.0-20210915214749-c084706c2272
go get: added golang.org/x/net v0.0.0-20210913180222-943fd674d43e
go get: added golang.org/x/sys v0.0.0-20210915083310-ed5796bab164
go get: added golang.org/x/term v0.0.0-20210503060354-a79de5458b56
go get: added golang.org/x/text v0.3.7
go get: added google.golang.org/protobuf v1.27.1
2022/01/22 04:10:15 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/jptosso/coraza-caddy 
go get: added github.com/antchfx/jsonquery v1.1.4
go get: added github.com/antchfx/xmlquery v1.3.6
go get: added github.com/antchfx/xpath v1.2.0
go get: added github.com/ip2location/ip2location-go/v9 v9.1.0
go get: added github.com/jptosso/aho-corasick v1.0.4
go get: added github.com/jptosso/coraza-caddy v1.1.0
go get: added github.com/jptosso/coraza-waf v1.1.0
go get: added github.com/oschwald/geoip2-golang v1.5.0
go get: added github.com/oschwald/maxminddb-golang v1.8.0
go get: added github.com/pcktdmp/cef v0.2.0
go get: upgraded github.com/sirupsen/logrus v1.7.0 => v1.8.1
go get: upgraded go.uber.org/atomic v1.7.0 => v1.9.0
go get: upgraded go.uber.org/multierr v1.6.0 => v1.7.0
2022/01/22 04:10:18 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/jptosso/coraza-pcre 
go get: added github.com/cloudflare/ahocorasick v0.0.0-20210425175752-730270c3e184
go get: added github.com/gijsbers/go-pcre v0.0.0-20161214203829-a84f3096ab3c
go get: added github.com/jptosso/coraza-pcre v0.0.0-20211201204829-1bea0f044896
go get: added github.com/jptosso/coraza-waf/v2 v2.0.0-beta.2.0.20211201203316-49869b6ff2ea
go get: upgraded go.uber.org/zap v1.19.0 => v1.19.1
2022/01/22 04:10:19 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/jptosso/coraza-libinjection 
go get: added github.com/jptosso/coraza-libinjection v0.0.0-20220109185528-78d7edbd2d9f
2022/01/22 04:10:19 [INFO] Build environment ready
2022/01/22 04:10:19 [INFO] Building Caddy
2022/01/22 04:10:19 [INFO] exec (timeout=0s): /usr/local/go/bin/go mod tidy 
2022/01/22 04:10:20 [INFO] exec (timeout=0s): /usr/local/go/bin/go build -o /root/caddy -ldflags -w -s -trimpath 
go build github.com/gijsbers/go-pcre: build constraints exclude all Go files in /root/go/pkg/mod/github.com/gijsbers/go-pcre@v0.0.0-20161214203829-a84f3096ab3c
# github.com/jptosso/coraza-libinjection
/root/go/pkg/mod/github.com/jptosso/coraza-libinjection@v0.0.0-20220109185528-78d7edbd2d9f/plugin.go:30:18: undefined: isSQLi
/root/go/pkg/mod/github.com/jptosso/coraza-libinjection@v0.0.0-20220109185528-78d7edbd2d9f/plugin.go:43:9: undefined: isXSS
2022/01/22 04:10:21 [INFO] Cleaning up temporary folder: /tmp/buildenv_2022-01-22-0410.3712383977
2022/01/22 04:10:21 [FATAL] exit status 2
```

**Comment by jptosso** [comment]
Hey, apparently you are using an old release, please use [9c2db28](https://github.com/jptosso/coraza-caddy/commit/9c2db28433e77d47c6e2d93c9acb6739b4ce8c89). coraza-caddy will be upgraded in a few days to coraza v2 stable.

**Comment by konstantin-921** [comment]
> Hey, apparently you are using an old release, please use [9c2db28](https://github.com/jptosso/coraza-caddy/commit/9c2db28433e77d47c6e2d93c9acb6739b4ce8c89). coraza-caddy will be upgraded in a few days to coraza v2 stable.

Ok. I'll try to install it after new release

**Comment by jptosso** [comment]
Coraza-caddy was updated to the latest version, feel free to try it

**Comment by konstantin-921** [comment]
Thanks I'll do it asap

**Comment by robgordon89** [comment]
Having the same issue with caddy and the latest.

Caddy Version: v2.4.6

```
xcaddy build --with github.com/jptosso/coraza-caddy
```

CaddyFile:
```
{
        order coraza_waf first
	log {
		output stdout
	}
}

:8080 {
    coraza_waf {
        directives `
            SecAction "id:1,pass,log"
        `
        include /etc/caddy/coraza/coraza.conf-recommended
        include /etc/caddy/coraza/coreruleset/crs-setup.conf.example
        include /etc/caddy/coraza/coreruleset/rules/*.conf
    }
    respond "Hello, world!"
}

```

results in 
```
... 'waf': provision http.handlers.waf: cannot load waf directives invalid profile path
```

**Comment by jptosso** [comment]
We have just created the v1.2.0 tag, which should fix the issue

**Comment by robgordon89** [comment]
ok seem to be getting over this first initial bug now 🤔 but running into another issue its likely with the CRS 

```
run: loading initial config: loading new config: loading http app module: provision http: server srv0: setting up route handlers: route 0: loading handler modules: position 0: loading module 'waf': provision http.handlers.waf: failed to compile rule (error parsing regexp: invalid or unsupported Perl syntax: `(?<`): FILES_NAMES|FILES "@rx (?<!&(?:[aAoOuUyY]uml)|&(?:[aAeEiIoOuU]circ)|&(?:[eEiIoOuUyY]acute)|&(?:[aAeEiIoOuU]grave)|&(?:[cC]cedil)|&(?:[aAnNoO]tilde)|&(?:amp)|&(?:apos));|['\"=]" "id:920120,phase:2,block,t:none,t:urlDecodeUni,msg:'Attempted multipart/form-data bypass',logdata:'%{MATCHED_VAR}',tag:'application-multi',tag:'language-multi',tag:'platform-multi',tag:'attack-protocol',tag:'paranoia-level/1',tag:'OWASP_CRS',tag:'capec/1000/210/272',ver:'OWASP_CRS/3.3.2',severity:'CRITICAL',setvar:'tx.anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
```

Also strangely any rules in CSR don't actually block the request just log 🤔

**Comment by jptosso** [comment]
There are two ways to fix that, you could update to the latest CRS version
(from yesterday, as they just removed PCRE requirements) or you can install
the plugin https://github.com/jptosso/coraza-pcre

If the rules are not blocking, that's probably because of your
SecDefaultAction settings, you should set: SecDefaultAction
"phase:2,deny,status:403,log"

Regards

**Comment by robgordon89** [comment]
Hey 👋 

oh that is what I was missing 👍 

Yeah I tried the latest nightly but still issues with regex.

```
run: loading initial config: loading new config: loading http app module: provision http: server srv0: setting up route handlers: route 0: loading handler modules: position 0: loading module 'waf': provision http.handlers.waf: failed to compile rule (error parsing regexp: invalid repeat count: `{0,100}`): REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "(?is)\r\n\w{1,50}\b\ (?:S(?:E(?:ARCH(?: CHARSET [\w\-_\.]{1,40})? (?:(KEYWORD \x5c)?(?:ALL|ANSWERED|BCC|DELETED|DRAFT|FLAGGED|RECENT|SEEN|UNANSWERED|UNDELETED|UNDRAFT|UNFLAGGED|UNSEEN|NEW|OLD)|(?:TEXT .{1,255}|TO .{1,255}|UID [0-9,:\*]+?|UNKEYWORD (?:\x5c(Seen|Answered|Flagged|Deleted|Draft|Recent)))|(?:BEFORE|ON|SENTBEFORE|SENTON|SENTSINCE|SINCE) \"?\d{1,2}-\w{3}-\d{4}\"?|(?:OR .{1,255} .{1,255}|SMALLER \d{1,20}|SUBJECT .{1,255})|(?:(?:BODY|CC|FROM)|HEADER .{1,100}) .{1,255}|(?:LARGER \d{1,20}|NOT .{1,255}|[0-9,:\*]+))|LECT [\w\"\.\-\x5c\/%\*&#]+)|T(?:ORE [0-9,:\*]+? [+-]?FLAGS(?:\.SILENT)? (?:\(\x5c[A-Za-z]{1,20}\)){0,100}|ARTTLS)|UBSCRIBE [\w\"\.\-\x5c\/%\*&#]+)|L(?:IST [\w\"~\-\x5c\/\*#\.]+? [\w\"\.\-\x5c\/%\*&#]+|OG(?:IN [A-Z0-9-_\.\@]{1,40} .*? |OUT))|C(?:(?:OPY [0-9,:\*]+|REATE) [\w\"\.\-\x5c\/%\*&#]+|APABILITY|HECK|LOSE)|RENAME [\w\"\.\-\x5c\/%\*&#]+? [\w\"\.\-\x5c\/%\*&#]+|UN(?:SUBSCRIBE [\w\"\.\-\x5c\/%\*&#]+|AUTHENTICATE)|EX(?:AMINE [\w\"\.\-\x5c%\*&#]+|PUNGE)|DELETE [\w\"\.\-\x5c%\*&#]+|FETCH [0-9,:\*]+|NOOP)" "id:932311,phase:2,block,t:none,t:escapeSeqDecode,msg:'Remote Command Execution: IMAP Command Execution',logdata:'Matched Data: %{TX.0} found within %{MATCHED_VAR_NAME}: %{MATCHED_VAR}',tag:'application-multi',tag:'platform-multi',tag:'attack-rce',tag:'paranoia-level/3',tag:'OWASP_CRS',tag:'capec/137/134',tag:'PCI/6.5.2',ctl:auditLogParts=+E,ver:'OWASP_CRS/3.4.0-dev',severity:'CRITICAL',setvar:'tx.rce_score=+%{tx.critical_anomaly_score}',setvar:'tx.anomaly_score_pl3=+%{tx.critical_anomaly_score}'"
```

I cant install the PCRE plugin with xcaddy it won't compile for some reason :( 

removing that rule from the release seems to work 👍 

``` 
/etc/caddy/coreruleset/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf
```

**Comment by jptosso** [comment]
Hey @robgordon89, sorry for the delay. the CRS team has solved that compatibility issue by replacing {0,100} with {,100}. It's some weird golang re2 bug.

**Comment by robgordon89** [comment]
hey @jptosso no probs i did notice that and it works as expected now, thanks for your help on this. I do have a weird bug that I am trying to replicate, but I will likely open another issue for this.

**Comment by jptosso** [comment]
Looking forward to it. Glad to help
